builder/dockerfile/builder.go

```diff
@@ -286,6 +293,14 @@ func (b *Builder) build(stdout io.Writer, stderr io.Writer, out io.Writer) (stri
 		return "", fmt.Errorf("No image was generated. Is your Dockerfile empty?")
 	}
 
+	if b.options.Squash {
+		var err error
+		b.image, err = b.docker.SquashImage(b.image, b.from.ImageID())
```

Comment:
It looks like `b.from` can be nil when doing `From scratch`. In that case, I think we want to just pass in `""` here.